src/Common/Config/AbstractConfigurationComparison.cpp

```diff
@@ -19,6 +19,56 @@ namespace
         result += subkey;
         return result;
     }
+
+    bool isSameConfigurationImpl(const Poco::Util::AbstractConfiguration & left, const String & left_key,
+                                 const Poco::Util::AbstractConfiguration & right, const String & right_key,
+                                 const std::unordered_set<std::string_view> * ignore_keys)
+    {
+        if (&left == &right && left_key == right_key)
+            return true;
+
+        /// Get the subkeys of the left and right configurations.
+        Poco::Util::AbstractConfiguration::Keys left_subkeys;
+        Poco::Util::AbstractConfiguration::Keys right_subkeys;
+        left.keys(left_key, left_subkeys);
+        right.keys(right_key, right_subkeys);
+
+        if (ignore_keys)
+        {
+            std::erase_if(left_subkeys, [&](const String & key) { return ignore_keys->contains(key); });
+            std::erase_if(right_subkeys, [&](const String & key) { return ignore_keys->contains(key); });
+        }
+
+        /// Check that the right configuration has the same set of subkeys as the left configuration.
+        if (left_subkeys.size() != right_subkeys.size())
+            return false;
+
+        if (left_subkeys.empty())
+        {
+            if (left.hasProperty(left_key))
+            {
+                return right.hasProperty(right_key) && (left.getRawString(left_key) == right.getRawString(right_key));
+            }
+            else
+            {
+                return !right.hasProperty(right_key);
+            }
+        }
+        else
```

Comment:
If subkeys are not empty, then we don't need to check the left_key & right_key property value is it?

Reply:
ignore this, I see its the configuration keys, so property value is only compared if there are no keys.

Reply:
Formally speaking there can be both keys and value at the same time:
```
<test>
    <a>5</a>
    <b>
        text
        <c>33</c>
        another text
        <d>44</d>
        final text
    </b>
</test>
```
however it would be really difficult to deal with such config while comparing and we don't use such things in our configurations anyway.

Reply:
I think I'll add a comment about that in the code.